Sources/Build/BuildPlan/BuildPlan.swift

```diff
@@ -654,6 +662,34 @@ public class BuildPlan: SPMBuildCore.BuildPlan {
         }
         return try description.symbolGraphExtractArguments()
     }
+
+    /// Returns the files and directories that affect the build process of this build plan.
+    public func inputs() -> [Input] {
```

Comment:
```suggestion
    package var inputs: [Input] {
```

Reply:
Accessing `inputs` is not O(1) because it iterates over all the root packages and that’s why I generally prefer to use a function over a computed property since a computed property access feels like an O(1) operation. Happy to use a computed property though if that matches SwiftPM’s style better.]

Somewhat related to https://www.swift.org/documentation/api-design-guidelines/#document-computed-property-complexity

Reply:
I don't think we've ever committed to computed properties always meaning `O(1)`. `String`'s `count` is `O(n)` and is still a computed property.

Reply:
OK, maybe it’s just me who has tried to keep computed properties at O(1) then (and probably not consistently either). Changed it to a computed property.